lib/components/Carousel.js

```diff
@@ -187,22 +187,39 @@ var Carousel = function (_Component) {
                     selectedItem: index
                 });
             }
+
+            // TODO: is onChange event emitter missing here?
         };
 
-        _this.handleOnChange = function (index, item) {
+        _this.handleOnChange = function (index, item, srcType) {
             if (_react.Children.count(_this.props.children) <= 1) {
                 return;
             }
 
-            _this.props.onChange(index, item);
+            _this.props.onChange(index, item, srcType);
```

Comment:
Instead of adding a new prop, what do you think about changing the signature of the `onChange` to include the source? Something along the lines of: `onChange: ({index: number, item: HTMLElement, source: 'click' | 'swipe' | 'autoPlay' }) => void`. This would introduce a breaking change and require a major bump but would also keep the API more future proof I'd say. Any thoughts?

Reply:
I think this could be an idea for a future major release, but for this new feature it isn't necessary yet.

(btw: you commented the generated file in lib, not the src)

Reply:
Please do not commit generated files, they will be generated on publishing. 
 
Also, I don't think we need to introduce a new prop to accomplish what you want... The suggestion I left above is an example on how to do it with the existing prop, with a small change in the API.

Reply:
Sorry, I don't understand, what you mean, maybe I need more text/explanation. Maybe I'm confused about 'prop'  when talking about onChange-'function-parameter'.

I don't think, we have to change the API because the third parameter in onChange would not break the API and 3 Parameters are imho OK for onChange. Only few developers will use the third parameter to differentiate between click or swipe. Most developers will use onChange for all changeEvents or onUserChange for the user triggered events.

I added onUserChange because I need a way to  differentiate user actions from programmatic willReceiveProps-actions.

I added srcType as third parameter to onChange and onUserChange because i think, there are use cases, where it is would be nice to have a 'prev'/'next' or 'click'/'swipe' or 'click-thumb'  info in onChange.